pkg/dataobj/internal/sections/streams/streams.go

```diff
@@ -33,6 +36,20 @@ type Stream struct {
 	Rows         int           // Number of rows in the stream.
 }
 
+func (s *Stream) Reset() {
+	s.ID = 0
+	s.Labels = nil
+	s.MinTimestamp = time.Time{}
+	s.MaxTimestamp = time.Time{}
+	s.Rows = 0
+}
+
+var streamPool = sync.Pool{
```

Comment:
this pool added about 10% more ops in the benchmark when re-using dataobjs between metastores. In reality we won't be reusing Streams very often (once per flush), so it might not be worth keeping this pool as the Stream objects will likely be deallocated between runs.
WDYT?